nautobot/extras/views.py

```diff
@@ -700,13 +700,16 @@ def post(self, request, class_path):
         elif form.is_valid():
             # Run the job. A new JobResult is created.
             commit = form.cleaned_data.pop("_commit")
+            # TODO: Define interface/API for getting queue (or other kwargs in future) for celery_kwargs
+            celery_kwargs = {}
```

Comment:
Check out how I constructed `celery_kwargs` from the incoming Job class in https://github.com/nautobot/nautobot/pull/1081/

I'd like to see if we can at least have a basic thing in the Job meta to pass these in for now? Or please let me know what your thoughts are here.

Reply:
My first thought is that I don't think we should implement a solution unless we have some level of confidence that it provides a good/usable/supportable interface. I have left out the implementation of celery_kwargs at this time b/c I have not got the feeling that there is consensus on this. I do think there has been agreement that we would like to support celery_kwargs, so this does add support for that feature. IMO, this is better than creating an implementation now, and changing the interface shortly thereafter.

As for moving that logic into the enqueue_job definition, that makes sense, this particular code was done to stick out and be obvious that something more needed to be done.

For an actual implementation that I would propose, I think there needs to be a get_celery_kwargs method that is supported by Job implementation. From a person understanding or typing code, I do not see a difference between these two:

Meta implementation

class MyJob(Job):
    Meta:
        celery_kwargs = {"queue": "my_queue"}
method implementation

class MyJob(Job):
    def get_celery_kwargs(self):
        return {"queue": "my_queue"}
However, implementing the method interface allows Job developers more flexibility in how celery_kwargs are defined, since they would have access to the Job form data.

class MyJob(Job):
    def get_celery_kwargs(self):
        if self.data["region"].slug == "Asia":
            queue = "east"
        else:
             queue = "west"
        return {"queue": queue}
This also follows the current implementation of Job subclasses where it is expected to have a run method, and supports test_* and and post_run methods that only accept self as an argument.

This is my opinion, if others disagree, then I am fine to implement whatever interface is desired at this time.

Reply:
I actually like both of the approaches you mentioned, so thank you for sharing them.

I think you’re raising a strong point to have the ability to pass `celery_kwargs` to `run_job` today and worry about the `Job` integration “tomorrow” as we finalize it. So that we’re not directly exposing a new interface that we haven’t yet fully thought out.